crates/agglayer-certificate-orchestrator/src/lib.rs

```diff
@@ -573,7 +586,68 @@ where
                     }
                 }
             }
-            _ => {}
+            Err(error) => {
+                warn!("Error during certification process: {}", error);
+                let certificate_error: Option<(CertificateId, CertificateStatusError)> = match error
+                {
+                    Error::ProofVerificationFailed {
+                        source,
+                        certificate_id,
+                    } => Some((certificate_id, source.into())),
+
+                    Error::ProverExecutionFailed {
+                        source,
+                        certificate_id,
+                    } => Some((
+                        certificate_id,
+                        CertificateStatusError::ProofGenerationError {
+                            generation_type: agglayer_types::GenerationType::Prover,
+                            source,
+                        },
+                    )),
+
+                    Error::NativeExecutionFailed {
+                        source,
+                        certificate_id,
+                    } => Some((
+                        certificate_id,
+                        CertificateStatusError::ProofGenerationError {
+                            generation_type: agglayer_types::GenerationType::Native,
+                            source,
+                        },
+                    )),
+
+                    Error::Types {
+                        source,
+                        certificate_id,
+                    } => Some((certificate_id, source.into())),
+
+                    Error::Storage(error) => {
+                        warn!(
+                            "Storage error happened in the certification process: {:?}",
+                            error
+                        );
+                        None
+                    }
+                    _ => None,
+                };
+
+                if let Some((certificate_id, error)) = certificate_error {
+                    if self
+                        .state_store
+                        .update_certificate_header_status(
+                            &certificate_id,
+                            &CertificateStatus::InError { error },
+                        )
+                        .is_err()
+                    {
+                        error!(
+                            "Certificate in error and failed to update the certificate header \
+                             status"
+                        );
```

Comment:
maybe also logging both errors: the `CertificateStatusError` and the error raised while attempting to update the certificate status

Reply:
The error is already logged above the match, currently in `warn` as I don't really know if all the errors are critical. We may move some to `error` if needed.